src/main/java/org/spongepowered/common/item/inventory/util/ItemStackUtil.java

```diff
@@ -105,8 +105,8 @@ public static ItemStack cloneDefensive(@Nullable ItemStack stack, int newSize) {
     }
 
     public static boolean compare(net.minecraft.item.ItemStack stack1, net.minecraft.item.ItemStack stack2) {
-        return net.minecraft.item.ItemStack.areItemStacksEqual(stack1, stack2);
-    }
+        return stack1.isItemEqual(stack2) && net.minecraft.item.ItemStack.areItemStackTagsEqual(stack1, stack2);
+    } // This ignores stack size on purpose
```

Comment:
Don't change the functionality of this method and keep the name the same. For a compare method, stack size DOES matter. If how this is used causes issues with stack size then rename this method. Your changes will lead to confusion down the line when an update ultimately has us touch this again.